src/Rules/Install.hs

```diff
@@ -42,41 +56,61 @@ installLibExecs = do
   libexecDir <- (-/- "bin") <$> ghclibdir
   dest <- destDir
   installDir (dest ++ libexecDir)
-  let programContexts = liftM2 programContext [minBound..] knownPackages  
-  forM_ programContexts $ \context@Context{..} -> do
-      activePackages <- interpretInContext context getPackages
-      when (stage == Stage0 || package == ghc) $ do
-          when (not (isLibrary package) && package `elem` activePackages) $ do
-              let bin = programInplaceLibPath -/- programName context <.> exe
-              need [bin]
-              installProgram bin (dest ++ libexecDir)
+  forM_ installStages $ \(pkg, mStgs) -> do
+    stgs <- mStgs
+    forM_ stgs $ \stg -> do
+      let context = programContext stg pkg
+      let bin = programInplaceLibPath -/- programName context <.> exe
+      need [bin]
+      installProgram bin (dest ++ libexecDir)
+      when (pkg == ghc) $ do
+        moveFile (dest ++ libexecDir -/- programName context <.> exe)
+                 (dest ++ libexecDir -/- "ghc" <.> exe)
+
+ghcSplitPkg :: Package
+ghcSplitPkg = utility "ghc-split"
+
+-- TODO: Consider Stage1Only
+installStages :: [(Package, Action [Stage])]
+installStages =
+  [ (ghc, maybeToList <$> latestBuildStage ghc)
+  , (ghcPkg, return [ Stage1 ])
+  , (ghcSplitPkg, return [ Stage0 ])
+  , (hp2ps, return [ Stage1 ])
+  , (hpc, return [ Stage1 ])
+  , (hsc2hs, return [ Stage1 ])
+  , (runGhc, return [ Stage1 ])
+  , (unlit, return [ Stage1 ]) ]
```

Comment:
> Can you give an example where using latestBuildStage instead of a hard-coded stage is wrong?

I don't have such an example, and not thinking that hardcoding is definitely the right way to go. I somehow got the impress that `dist-install` suffix means `Stage1`, from code like https://github.com/snowleopard/hadrian/blob/c4e2e45be2e11e5785b033ab10511455c2dd00fc/src/Rules/Program.hs#L26 etc.

But well, I guess `latestBuildStage` might make more sense here.